### PR TITLE
Feat: basic auth, wishList and Cart with Functionality

### PR DESCRIPTION
**Pull request by sainath7878**
Files to check

- [src/components/cart/Amount.jsx](https://github.com/sainath7878/joddkart/pull/4/files#diff-a26bf00677014a4c148a06ef885723e8913a19f9bd9e47ae0d8e7a89211627f7)
- [src/components/cart/Cart.jsx](https://github.com/sainath7878/joddkart/pull/4/files#diff-cbe3f21a64c37da821ecb69771ac4aaa6f5f539678becb2cb4a9c9aa40cde4a0)
- [src/components/cart/cartCard.jsx](https://github.com/sainath7878/joddkart/pull/4/files#diff-0760ebc3ddd5e8a80f01436e84c9a3e84df7e58af60956cfd74800c28bb9c898)
- [src/components/productCard/productCard.jsx](https://github.com/sainath7878/joddkart/pull/4/files#diff-290ad800d568759e183ea48235acffdbd12927170cbbba01e9c4382eb273d289)
- [src/components/wishList/wishList.jsx](https://github.com/sainath7878/joddkart/pull/4/files#diff-aee63bde066fcd38397b57242df20ede392507c8033c5aa0f420ddd4d7e9b1c3)
- [src/reducer/filterReducer.js](https://github.com/sainath7878/joddkart/pull/4/files#diff-9fd257b002aed8365b59bd6b2ddb3c5b491f772ce957bb2fc5fa82ce83ddcf65)
- [src/pages/wishlistPage/WishListPage.jsx](https://github.com/sainath7878/joddkart/pull/4/files#diff-2f1016911795b2f6f7c84662fcc4118db42e40e0950a9d1023d2e2066662388f)
- [src/pages/CartPage/CartPage.jsx](https://github.com/sainath7878/joddkart/pull/4/files#diff-25f2e824ed86e7a3d77b6ef05040caab10bc7dd8e17359d537d187d6ecd579c5)

### Thanks for reviewing!

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+REACT_APP_JWT_SECRET = "joddKartOP"
```
Comment (anik31) [inline .env:1]: suggestion - don't push .env files to github, add it to gitignore.

File: src/components/cart/Cart.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+function Cart() {
+  const { state } = useProducts();
+  const { cart, loader } = state;
+
+
+  return (
+    <>
+      {loader ? (
+        <h1>Loading Data!!!</h1>
```
Comment (anik31) [inline src/components/cart/Cart.jsx:12]: appreciation - good that you're showing loader

**Review by anik31 — APPROVED** [review]
Good work 💯 Everything looks good. The functionality is working fine in the preview.

suggestion - you can breakdown the larger code files into small files (you can do it later).

File: src/components/cart/Amount.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        (acc, {price, qty}) => ({
+          ...acc,
+          totalMRP: acc.totalMRP + price.original * qty,
+          totalDiscount:
+            acc.totalDiscount + price.discounted * qty,
+        }),
+        { totalMRP: 0, totalDiscount: 0 }
+      )
+    );
+  }, [cart]);
```
Comment (Maianki) [inline src/components/cart/Amount.jsx:21]: Suggestion: This doesn't look like a side-effect so you can do this in a simple function.

**Review by Maianki — APPROVED** [review]
Looks good to merge. 
Things I liked about your code:
- Great folder structure everything is kept separate.
- Separate files for reducer and context.

File: .env (hunk 1 above)
Comment (sainath7878) [reply .env:1]: forgot to put this in git ignore. Thanks for pointing out!

File: src/components/cart/Amount.jsx (hunk 3 above)
Comment (sainath7878) [reply src/components/cart/Amount.jsx:21]: I wanted to change my amount every time my cart item changes. So I put it in useEffect and gave the dependency of cart.

**Pull request merged**